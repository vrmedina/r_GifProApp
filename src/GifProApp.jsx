import { useState } from "react";
import {AddCategory, GifGrid} from './components';

export const GifProApp = () => {
    const [categories, setCategories] = useState([]);

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories]);
    }
    // categories.push(newCategory)
    // setCategories(['Agregada', ...categories]) 
    // const onAddCategory = () => {setCategories( c => [ ...c, 'Agregada'])};

    return (
    <>
        <h1>GifProApp</h1>

        <AddCategory 
            onNewCategory={onAddCategory}
        />

        {
            categories.map(category => ( 
                <GifGrid 
                    key={category}
                    category={category}
                    limit={6}
                /> 
            ))
        }
    </>
    );
}