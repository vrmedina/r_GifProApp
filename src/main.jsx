import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifProApp } from './GifProApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifProApp/>
  </React.StrictMode>
);