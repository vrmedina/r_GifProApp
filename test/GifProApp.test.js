import { fireEvent, render, screen } from "@testing-library/react";
import { GifProApp } from "../src/GifProApp";

describe('Pruebas en <GifProApp/>', () => {
    test('debe coincidir con el snapshot', () => {
        const { container } = render( <GifProApp /> );
        expect(container).toMatchSnapshot();
    });

    test('debe agregar una nueva categoria', () => {
        render( <GifProApp /> );
        const input = screen.getByRole('textbox'); 
        const form = screen.getByRole('form'); 

        fireEvent.input(input, { target: { value: 'Vegeta' } });
        fireEvent.submit(form);

        expect( screen.getByText('Goku') )
        expect( screen.getByText('Vegeta') )
    });

    test('no debe agregar una categoria que ya exista', () => {
        render( <GifProApp /> );
        const input = screen.getByRole('textbox'); 
        const form = screen.getByRole('form'); 

        fireEvent.input(input, { target: { value: 'Goku' } });
        fireEvent.submit(form);

        fireEvent.input(input, { target: { value: 'Goku' } });
        fireEvent.submit(form);

        expect(screen.getAllByRole('heading', { level: 3 }).length).toBe(1);
    });
});