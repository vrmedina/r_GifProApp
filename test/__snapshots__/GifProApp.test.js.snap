// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifProApp/> debe coincidir con el snapshot 1`] = `
<div>
  <h1>
    GifProApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
    <button>
      Buscar
    </button>
  </form>
  <h3>
    Goku
  </h3>
  <h2>
    Cargando tus Gifs...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
